### PR TITLE
RHIROS-1112 - process systems on update event without platform_metadata

### PR DESCRIPTION
**Pull request by upadhyeammit**
## Documentation update? :memo:

- [ ] Yes
- [ ] No

## Security Checklist :lock:

Upon raising this PR please go through [RedHatInsights/secure-coding-checklist](https://github.com/RedHatInsights/secure-coding-checklist)

## :guardsman: Checklist :dart:

- [x] Bugfix
- [ ] New Feature
- [ ] Refactor
- [ ] Unittests Added
- [ ] DRY code
- [ ] Dependency Added
- [ ] DB Migration Added

## Additional :mega:

Feel free to add any other relevant details such as __links, notes, screenshots__, here.

### CODE REVIEW

File: ros/processor/inventory_events_consumer.py [hunk 1]
```diff
@@ -87,6 +87,12 @@ def run(self):
         LOG.warning("Stopping inventory consumer")
         self.consumer.close()
 
+    def update_event_without_platform_metadata(self, msg):
+        if msg["platform_metadata"] is None and msg["type"] == "updated" \
+                and msg["host"]["system_profile"]["cloud_provider"] == "aws":
```
Comment (upadhyeammit) [inline ros/processor/inventory_events_consumer.py:92]: I am not sure if I should gate on "aws" specifically here, we dont use to do it before!

**Review by saltgen — APPROVED** [review]
LGTM :+1:

File: ros/processor/inventory_events_consumer.py [hunk 2]
```diff
@@ -119,42 +119,67 @@ def process_system_details(self, msg):
...
-                ).inc()
-                LOG.info(
-                    f"{self.prefix} - Refreshed system {system.inventory_id} ({system.id}) "
-                    f"belonging to account: {account.account} ({account.id}) and org_id: {account.org_id}"
-                )
-            except Exception as err:
-                processor_requests_failures.labels(
-                    reporter=self.reporter, org_id=account.org_id
-                ).inc()
+                system = update_system_record(db.session, **system_fields)
```
Comment (kgaikwad) [inline ros/processor/inventory_events_consumer.py:131]: There is a scenario where system already been registered and enabled ros later. In such scenario inventory service will send out message with `updated` event type as system already exists on Inventory side.  With this gate condition, ROS will reject such systems.
This needs to be handled.

File: ros/lib/utils.py [hunk 3]
```diff
@@ -255,6 +267,18 @@ def system_allowed_in_ros(msg, reporter):
...
         cloud_provider = msg["results"]["system"]["metadata"].get('cloud_provider')
     elif reporter == 'INVENTORY EVENTS':
-        is_ros = msg["platform_metadata"].get("is_ros")
         cloud_provider = msg['host']['system_profile'].get('cloud_provider')
+        if msg.get('type') == 'updated':
+            updated_event = True
+        else:
+            updated_event = False
+        if updated_event:
+            return is_valid_cloud_provider(cloud_provider)
```
Comment (kgaikwad) [inline ros/lib/utils.py:276]: I didn't understand why `cloud_provider` check is ONLY applicable for updated event?

**Pull request merged**